### DISCUSSION TITLE
Preview mode source tabs

### DISCUSSION THREAD

**Discussion by jaril** [RecordReplay/devtools#4665] [Ideas]
In the past we've found that to use Replay well, you had to switch back and forth between Replay and your code editor. Since then, we've tried to reduce the need to switch to the code editor with things like full text search. This makes it easier to explore the project and look through files while searching, but it also introduced a new problem — tab bloat. 

VS Code solves this by having preview mode (?) tabs. A preview mode tab disappears when you open another tab, unless you perform some explicit action indicating that you'd like the tab to be opened properly (e.g. double click, or editing the file). If we're expecting a similar skimming-through-many-files workflow, we probably should implement something similar.

**Comment by jonbell-lot23** [comment]
March 15, 2022:

This topic is coming up more now that we have different layouts. Right now, "pin to bottom" and "pin to left" hide new tabs entirely, so it can feel broken when you click on something and nothing appears.

Jaril and I discussed this. My preference at this point is to at least flash the sources tab to people are aware that a tab opened behind the scenes. But it's possible we'll want to think more about tab bloat in this scenario, as mentioned here.
